### PR TITLE
Reworked display of product alerts

### PR DESCRIPTION
**Pull request by luigifab**
### Description

Same but different :)

_for configuration (system > configuration > catalog):_

cnf-before

cnf-after

_for product view:_

prd-before

prd-after

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All automated tests passed successfully (all builds are green)
 - [x] Add yourself to contributors list

### CODE REVIEW

File: app/code/core/Mage/ProductAlert/etc/system.xml [hunk 1]
```diff
@@ -25,52 +25,67 @@
...
                             <sort_order>1</sort_order>
                             <show_in_default>1</show_in_default>
                             <show_in_website>1</show_in_website>
                             <show_in_store>1</show_in_store>
-                        </allow_price>
-                        <allow_stock translate="label">
-                            <label>Allow Alert When Product Comes Back in Stock</label>
+                        </email_identity>
+                        <head_price>
+                            <label>Product price alert</label>
```
Comment (kiatng) [inline app/code/core/Mage/ProductAlert/etc/system.xml:38]:
```suggestion
                            <label>Allow Alert When Product Price Changes</label>
```

File: app/code/core/Mage/ProductAlert/etc/system.xml [hunk 2]
```diff
@@ -25,52 +25,67 @@
...
-                            <sort_order>2</sort_order>
+                            <sort_order>12</sort_order>
                             <show_in_default>1</show_in_default>
                             <show_in_website>1</show_in_website>
                             <show_in_store>1</show_in_store>
                         </email_price_template>
-                        <email_stock_template translate="label">
-                            <label>Stock Alert Email Template</label>
+                        <head_stock>
+                            <label>Product stock alert</label>
```
Comment (kiatng) [inline app/code/core/Mage/ProductAlert/etc/system.xml:64]:
```suggestion
                            <label>Allow Alert When Product Comes Back in Stock</label>
```

**Review by kiatng — COMMENTED** [review]
I like it. But how about retaining the original labels, which are more explanatory?

**Comment by S0FTWEX** [comment]
I would like to keep the original labels too (they are more explanatory). Or better: you can move them to <comment></comment>  -  this is the best solution.

**Comment by luigifab** [comment]
I'm going to update soon.

File: app/code/core/Mage/Adminhtml/Block/Catalog/Product/Edit/Tab/Alerts/Price.php [hunk 3]
```diff
@@ -30,7 +30,7 @@ public function __construct()
         $this->setDefaultDir('desc');
         $this->setUseAjax(true);
         $this->setFilterVisibility(false);
-        $this->setEmptyText(Mage::helper('catalog')->__('There are no customers for this alert'));
+        $this->setEmptyText(Mage::helper('catalog')->__('There are no customers for this alert.'));
```
Comment (fballiano) [inline app/code/core/Mage/Adminhtml/Block/Catalog/Product/Edit/Tab/Alerts/Price.php:33]: I think I wouldn't break the translation packs just to add the dot :-(

Comment (luigifab) [reply app/code/core/Mage/Adminhtml/Block/Catalog/Product/Edit/Tab/Alerts/Price.php:33]: The translation already exists.

Comment (fballiano) [reply app/code/core/Mage/Adminhtml/Block/Catalog/Product/Edit/Tab/Alerts/Price.php:33]: true, didn't notice, sorry!

**Comment by ADDISON74** [comment]
I agree with this PR in the following design

**1. System > Configuration > Catalog**

screenshot_1

**Comment**: we are in the "Product Alerts" section, the two sub-sections must be "Price" and "Back in Stock", as obvious and simple as possible, without abusing of words. The rest of the phrases for every field remained the initial ones.

**2. Product Edit page > Product Alerts tab**

screenshot_2

**Commend**: Initial phrases "Price alert subscription was saved." and "Stock notification was saved." they are devoid of logic. Those grids are just a listing of alerts, but they can't be edited to talk about a save.

I would rename them as "Price" and "Back in Stock" since we are in the "Price Alerts" tab, again without abusing of words. We can reformulate them "Price Notifications" and "Back in Stock Notifications" as well or "Price Subscriptions" and "Back in Stock Subscriptions".

**Comment by ADDISON74** [comment]
Some personal opinions:

1. I agree with the removal of the <div> container containing h4 - "Product Alerts" from the /product/tab/alert.phtml file. Its display is illogical and does not comply with the Backend presentation rules.

2. I do not agree with the two <div> containers inserted in the /Tab/Alerts.php file. I would like to keep the presentation rules as before, if we make this change it would mean making changes in other places as well. Changing the titles is enough.

**Comment by fballiano** [comment]
tested again today, I like this PR but I too would remove the div around the table as @ADDISON74 says.

**Review by fballiano — DISMISSED** [review]
I've removed those divs around the table and IMHO it could be merged now :-)

**Comment by ADDISON74** [comment]
@fballiano - Let's not rush because it's not finished yet.  Please take a look here https://github.com/OpenMage/magento-lts/pull/3303#issuecomment-1634851225. I made important observations after testing that must be implemented.

File: app/code/core/Mage/Adminhtml/Block/Catalog/Product/Edit/Tab/Alerts.php [hunk 4]
```diff
@@ -44,15 +44,15 @@ protected function _prepareLayout()
 
         if ($alertPriceAllow) {
             $accordion->addItem('price', [
-                'title'     => Mage::helper('adminhtml')->__('Price alert subscription was saved.'),
-                'content'   => $this->getLayout()->createBlock('adminhtml/catalog_product_edit_tab_alerts_price')->toHtml() . '<br />',
+                'title'     => Mage::helper('adminhtml')->__('Product price alert'),
```
Comment (ADDISON74) [inline app/code/core/Mage/Adminhtml/Block/Catalog/Product/Edit/Tab/Alerts.php:47]: 1. This should be changed to "Price Alerts". Using Product as prefix is like an abuse in my opinion
2. It needs a row for translation in Mage_Catalog.csv

File: app/code/core/Mage/Adminhtml/Block/Catalog/Product/Edit/Tab/Alerts.php [hunk 5]
```diff
@@ -44,15 +44,15 @@ protected function _prepareLayout()
...
+                'content'   => $this->getLayout()->createBlock('adminhtml/catalog_product_edit_tab_alerts_price')->toHtml(),
                 'open'      => true
             ]);
         }
         if ($alertStockAllow) {
             $accordion->addItem('stock', [
-                'title'     => Mage::helper('adminhtml')->__('Stock notification was saved.'),
-                'content'   => $this->getLayout()->createBlock('adminhtml/catalog_product_edit_tab_alerts_stock'),
+                'title'     => Mage::helper('adminhtml')->__('Product stock alert'),
```
Comment (ADDISON74) [inline app/code/core/Mage/Adminhtml/Block/Catalog/Product/Edit/Tab/Alerts.php:54]: 1. This should be changed to "Back in Stock Alerts". Using Product as prefix is like an abuse in my opinion. Also I keep the correct description from configuration
2. It needs a row for translation in Mage_Catalog.csv

File: app/code/core/Mage/ProductAlert/etc/system.xml [hunk 6]
```diff
@@ -25,52 +25,67 @@
...
+                            <source_model>adminhtml/system_config_source_email_identity</source_model>
                             <sort_order>1</sort_order>
                             <show_in_default>1</show_in_default>
                             <show_in_website>1</show_in_website>
                             <show_in_store>1</show_in_store>
-                        </allow_price>
-                        <allow_stock translate="label">
-                            <label>Allow Alert When Product Comes Back in Stock</label>
+                        </email_identity>
+                        <head_price>
```
Comment (ADDISON74) [inline app/code/core/Mage/ProductAlert/etc/system.xml:37]: Missing ```translate="label"```

File: app/code/core/Mage/ProductAlert/etc/system.xml [hunk 7]
```diff
@@ -25,52 +25,67 @@
...
+                            <label>Allow Alert When Product Price Changes</label>
+                            <frontend_model>adminhtml/system_config_form_field_heading</frontend_model>
+                            <sort_order>10</sort_order>
+                            <show_in_default>1</show_in_default>
+                            <show_in_website>1</show_in_website>
+                            <show_in_store>1</show_in_store>
+                        </head_price>
+                        <allow_price translate="label">
+                            <label>Enabled</label>
```
Comment (ADDISON74) [inline app/code/core/Mage/ProductAlert/etc/system.xml:46]: Enabled needs a row in the translation file Mage_ProductAlert.csv.

File: app/code/core/Mage/ProductAlert/etc/system.xml [hunk 8]
```diff
@@ -25,52 +25,67 @@
...
                             <source_model>adminhtml/system_config_source_email_template</source_model>
-                            <sort_order>2</sort_order>
+                            <sort_order>12</sort_order>
                             <show_in_default>1</show_in_default>
                             <show_in_website>1</show_in_website>
                             <show_in_store>1</show_in_store>
                         </email_price_template>
-                        <email_stock_template translate="label">
-                            <label>Stock Alert Email Template</label>
+                        <head_stock>
```
Comment (ADDISON74) [inline app/code/core/Mage/ProductAlert/etc/system.xml:63]: Missing translate="label".

File: app/code/core/Mage/ProductAlert/etc/system.xml (hunk 6 above)
Comment (fballiano) [inline app/code/core/Mage/ProductAlert/etc/system.xml:37]:
```suggestion
                        <head_price translate="label">
```

File: app/code/core/Mage/ProductAlert/etc/system.xml [hunk 9]
```diff
@@ -25,52 +25,67 @@
...
```
Comment (fballiano) [inline app/code/core/Mage/ProductAlert/etc/system.xml:63]:
```suggestion
                        <head_stock translate="label">
```

File: app/code/core/Mage/Adminhtml/Block/Catalog/Product/Edit/Tab/Alerts.php (hunk 4 above)
Comment (luigifab) [reply app/code/core/Mage/Adminhtml/Block/Catalog/Product/Edit/Tab/Alerts.php:47]: No. Don't add new translation for an useless thing.

**Pull request closed without merging**